src/client/cli/cmd/remote_settings_handler.cpp

```diff
@@ -0,0 +1,201 @@
+        : RemoteSettingsCmd{stub, term} // need to ensure refs outlive this
+    {
+        mp::SetRequest set_request;
+        set_request.set_verbosity_level(verbosity);
+        set_request.set_key(key.toStdString());
+        set_request.set_val(val.toStdString());
+
+        [[maybe_unused]] auto ret = dispatch(&RpcMethod::set, set_request, on_success<mp::SetReply>, on_failure);
+        assert(ret == mp::ReturnCode::Ok && "should have thrown otherwise");
+    }
+};
+
+class RemoteKeys : public RemoteSettingsCmd
+{
+public:
+    RemoteKeys(QString fallback, mp::Rpc::StubInterface& stub, mp::Terminal* term, int verbosity)
+        : RemoteSettingsCmd{stub, term}
+    {
+        mp::KeysRequest keys_request;
+        keys_request.set_verbosity_level(verbosity);
+
+        auto custom_on_success = [this](mp::KeysReply& reply) {
+            for (auto& key : *reply.mutable_settings_keys())
+                keys.insert(QString::fromStdString(std::move(key)));
+
+            return mp::ReturnCode::Ok;
+        };
+
+        auto custom_on_failure = [this, fallback = std::move(fallback)](grpc::Status& status) mutable {
+            if (status.error_code() == grpc::StatusCode::NOT_FOUND)
+            {
+                keys.insert(std::move(fallback));
+                return mp::ReturnCode::Ok;
+            }
+
+            return on_failure(status); // return in all branches, even though this (currently) throws
+        };
+
+        [[maybe_unused]] auto ret = dispatch(&RpcMethod::keys, keys_request, custom_on_success, custom_on_failure);
+        assert(ret == mp::ReturnCode::Ok && "should have thrown otherwise");
+    }
+
+public:
+    std::set<QString> keys = {};
+};
+} // namespace
+
+mp::RemoteSettingsHandler::RemoteSettingsHandler(QString key_prefix, mp::Rpc::StubInterface& stub,
+                                                 multipass::Terminal* term, int verbosity)
+    : key_prefix{std::move(key_prefix)}, stub{stub}, term{term}, verbosity{verbosity}
+{
+    assert(term);
+}
+
+QString mp::RemoteSettingsHandler::get(const QString& key) const
+{
+    if (key.startsWith(key_prefix))
+    {
+        assert(term);
+        return RemoteGet{key, stub, term, verbosity}.got;
+    }
+
+    throw mp::UnrecognizedSettingException{key};
+}
+
+void mp::RemoteSettingsHandler::set(const QString& key, const QString& val)
+{
+    if (key.startsWith(key_prefix))
+    {
+        assert(term);
+        RemoteSet(key, val, stub, term, verbosity);
+    }
+    else
+        throw mp::UnrecognizedSettingException{key};
+}
+
+std::set<QString> mp::RemoteSettingsHandler::keys() const
+{
+    assert(term);
+
+    auto fallback = QStringLiteral("%1* \t (need daemon to find out actual keys)").arg(key_prefix);
+    return RemoteKeys{std::move(fallback), stub, term, verbosity}.keys;
+}
+
+mp::RemoteHandlerException::RemoteHandlerException(grpc::Status status)
+    : std::runtime_error{"Error reaching remote setting"}, status{std::move(status)}
```

Comment:
But this is caused by problems other than reaching the daemon. The word "reaching" was an attempt at substituting for both reading and writing (the setting), but I agree not a very successful one at that. How about "Error accessing remote setting"?

NB: this message is not actually shown to the user, right now. The exception is [caught at client.cpp](https://github.com/canonical/multipass/blob/40c4115d41c0b68cd91ff95a52ce2e25acaefbfd/src/client/cli/client.cpp#L143) and the status code is translated into an error as usual. But it could be, if called from somewhere else (e.g. GUI), so we can't make assumptions and need to provide a proper message anyway.